### ISSUE TITLE
Beginning yesterday it fails with INTERNAL SERVER ERROR

### ISSUE THREAD

**Issue by yurivict** [oldlaptop/iheart-mplayer#2]
Every station fails: urllib.error.HTTPError: HTTP Error 500: INTERNAL SERVER ERROR

**Comment by yurivict** [comment]
This problem is intermittent, also can exist for some stations and not for others.
Might as well be iheart server problem, but in any case iheart-mplayer usability is definiteky impacted.

**Comment by oldlaptop** [comment]
I can't reproduce this here, right now. Can you post some specific station IDs that you've had trouble with? This problem is indeed probably caused by some hiccup in iheart's API server, but it also could be fallout from the API change they made a few months ago.

**Comment by russcrisp** [comment]
I just discovered this application. I loaded it up, played a couple of stations, then the HTTP Error 500: INTERNAL SERVER ERROR manifested itself. Even on stations where the station number is known to be invalid. Waited a few minutes, and tried again, then it worked like a charm.

I've noticed that playing the rtmp URLs in mplayer directly aren't always reliable. Sometimes it works, sometimes it doesn't..

I hope we can figure this out because this is really a good app.  I'll volunteer to help code, debug, etc, in whatever way I can.

**Comment by yurivict** [comment]
This happened intermittently for me. I don't think this is station specific.
Might as well be the server fault.

**Comment by russcrisp** [comment]
agreed.. It's hard to pinpoint.. I just played a stream from WGST in Atlanta for 3 hours, and it was flawless. I'm going to study the code a little and see if I can understand it.

**Comment by oldlaptop** [comment]
As of cddde77, it should retry any failed HTTP request three times automatically. Given that the problem seems to be with iheart's servers, this is probably the best that can be done.

**Comment by yurivict** [comment]
It didn't seem to be intermittent though, it just kept failing for particular station no matter what, until it gets to normal.

**Comment by oldlaptop** [comment]
That's odd. I just saw the issue here, finally, exactly once (tried again a few _seconds_ later and it worked fine). I'll ask again, do you have an example station that you've had issues with? My own usage (and therefore testing) of this is pretty narrow, there's a good chance there's something odd about the stations _you_ use that I don't know about.

**Comment by yurivict** [comment]
Yes, I had a problem with station numbers 2285 and 5953, but now they both work. At some point one of them (2285) didn't work several times in a row, but other stations worked fine. Then I tried number 2285 again, and it still didn't work several times in a row. This is why I don't think the problem is intermittent in a short period of time, only in the longer term.

I know this doesn't make it easier to understand the problem, but this is what I observed. And it looks like this is the server problem.
